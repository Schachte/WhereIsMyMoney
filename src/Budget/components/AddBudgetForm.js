import React, { Component } from 'react';
import { Field, reduxForm, SubmissionError } from 'redux-form/immutable';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import 'react-datepicker/dist/react-datepicker.css';

class AddBudgetForm extends Component {

  constructor (props) {
    super(props)
    this.state = {
      startDate: moment()
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(date) {
    this.setState({
      startDate: date
    });
  }

  categoryInput({input, type, label, meta: { touched, error }, ...custom}) {
    const hasError = touched && error !== undefined;
    const hasLabel = label !== undefined;
    return (
      <div style={{marginTop: '15px'}}>
        {hasLabel && <label>{label}</label>}
        <input
          id={label}
          className='form-control'
          {...input}
          {...custom}
        />
      </div>
    )
  }

  renderDatePicker({input, label, placeholder, meta: {touched, error} }) {
    const hasLabel = label !== undefined;
    return (
      <div>
        {hasLabel && <label>{label}</label>}
        <DatePicker id={label} className='form-control' {...input} value = {this.state.startDate.format('DD')} onChange={this.handleChange} />
        {touched && error && <span>{error}</span>}
      </div>
    )
  };

  submit(dataValues) {
    const formData = dataValues.toJS();
    console.log(formData)
    this.props.addBudget({
      budgetName: formData.budgetCategory,
      monthlyCost: formData.budgetMonthlyCost,
      rollOverEnabled: formData.budgetRollover = (formData.budgetRollover == 'true'),
      dueDate: this.state.startDate.format('DD').toString()
    });
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <div className='form-group'>
        <form onSubmit={handleSubmit(this.submit.bind(this))}>
          <Field name='budgetCategory' component={this.categoryInput} placeholder="Enter Category of Budget" label="Budget Category"/>
          <Field name='budgetMonthlyCost' component={this.categoryInput} placeholder="Enter Monthly Budget Cost" label="Monthly Budget: "/>
          <div style={{marginTop: '10px'}}>
            <label>Budget Rollover</label><br/>
            <label>
              <Field name='budgetRollover' checked="checked" component='input' type="radio" value={"true"} label="Budget Rollover Next Month" />
              {' '}
              Enable Budget Rollover
            </label>
            <label>
              <Field name='budgetRollover' component='input' type="radio" value={"false"} label="Budget Rollover Next Month" />
              {' '}
              Disable Budget Rollover
            </label>
          </div>
          <Field name='budgetDateDue' component={this.renderDatePicker.bind(this)} label="Day of Month Due" />
          <br/>
          <button type='submit' className='btn btn-primary'>Submit</button>
        </form>
      </div>
    )
  }
}

const validate = (values) => {
  const errors = {};
  values = values.toJS();

  if (!values.budgetCategory || values.budgetCategory.trim() === '') {
    errors.budgetCategory = 'Budget Category Required!';
  }

  return errors;
}

export default reduxForm({
  form: 'budgetForm',
  validate
})(AddBudgetForm);
