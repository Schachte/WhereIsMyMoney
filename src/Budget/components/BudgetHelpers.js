export function makeFieldEditable(fieldData, updateFieldEnableRequest, editable) {
  updateFieldEnableRequest({
    editable: editable,
    fieldName: fieldData.budgetName
  })
}
export default {
  makeFieldEditable
};
