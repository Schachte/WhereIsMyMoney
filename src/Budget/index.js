import Immutable from 'immutable'
import { createAction } from 'redux-actions';
import axios from 'axios';
import moment from 'moment';

/**************
INITIAL STATE
***************/
const INITIAL_STATE = Immutable.fromJS({
  budgetCategories: [
    {budgetName: 'budgetName', monthlyCost: '192', rollOverEnabled: "rollOverDisabled", dueDate: '02'}
  ],
  budgetFormEditable: {errors: [{budgetNameError: false, monthlyCostError: false}] },
});

/**************
TYPES
***************/
export const ADD_BUDGET = 'src/Budget/ADD_BUDGET';
export const ADD_EDITABLE_FIELD_ERRORS = 'src/Budget/ADD_EDITABLE_FIELD_ERRORS';
export const UPDATE_BUDGET_ENTRY = 'src/Budget/UPDATE_BUDGET_ENTRY';

/**************
REDUCER LOGIC FLOW
***************/
export default function (state = INITIAL_STATE, action) {
  switch (action.type) {
    case ADD_BUDGET:
      return state.updateIn(['budgetCategories'], arr => arr.push(Immutable.fromJS(action.payload)))

    case ADD_EDITABLE_FIELD_ERRORS:
      return state.setIn(['budgetFormEditable', 'errors', 0], Immutable.fromJS(action.payload))

    case UPDATE_BUDGET_ENTRY:
      return state.setIn(
        [
          'budgetCategories',
          action.payload.editedStateIndex,
        ], Immutable.fromJS(action.payload.newBudget));

    default:
      return state;
  }
}

/**************
ACTIONS CREATORS
***************/
export const addBudget = createAction(ADD_BUDGET);
export const addEditableFieldErrors = createAction(ADD_EDITABLE_FIELD_ERRORS);
export const updateBudgetEntry = createAction(UPDATE_BUDGET_ENTRY);

/**************
ACTION REQUESTS
***************/
//TODO: Honestly, this is pretty unnecessary as I am not resolving promises
export function addBudgetCategoryRequest(data) {
  return (dispatch) => {
    dispatch(addBudget(data));
  }
}
