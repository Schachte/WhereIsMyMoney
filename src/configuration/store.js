import {createStore, applyMiddleware, compose} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducers';

export function configureStore(){
  // Use redux dev tools if available, otherwise use default composition;
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const store = createStore(reducers, composeEnhancers(
    applyMiddleware(thunk)
  ));

  return store;
}
