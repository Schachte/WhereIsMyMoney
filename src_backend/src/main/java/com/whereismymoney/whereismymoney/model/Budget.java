package com.whereismymoney.whereismymoney.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Budget {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    String budgetName;
    Double budgetCost;
    int budgetDate;

    public Budget() {}

    public Budget(Long id, String budgetName, Double budgetCost, int budgetDate, Date nextNotification) {
        this.id = id;
        this.budgetName = budgetName;
        this.budgetCost = budgetCost;
        this.budgetDate = budgetDate;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getBudgetName() {
        return budgetName;
    }

    public void setBudgetName(String budgetName) {
        this.budgetName = budgetName;
    }

    public Double getBudgetCost() {
        return budgetCost;
    }

    public void setBudgetCost(Double budgetCost) {
        this.budgetCost = budgetCost;
    }

    public int getBudgetDate() {
        return budgetDate;
    }

    public void setBudgetDate(int budgetDate) {
        this.budgetDate = budgetDate;
    }
}
