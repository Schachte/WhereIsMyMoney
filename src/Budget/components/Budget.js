import React, { Component } from 'react';
import AddBudgetForm from './AddBudgetForm';

const metaProperties = {
  pageTitle: 'Review Current Budget',
  budgetCardTitle: 'Add A New Budget Item'
}

const budgetWidgets = [
  'Monthly Budget',
  'Current Status',
  'Amount Left',
  'Avg. Weekly Spent'
];

const BudgetWidgetCard = (props) => {
  return (
    <div className="col-1 col-sm-3 placeholder" style={{float: 'left'}} key={`CARD_${props.i}`}>
      <div className="card card-inverse" style={{backgroundColor: "#333", borderColor: "#333", color: "#FFF", padding: "10px"}}>
        <div className="card-block">
          <h4 className="card-title">{props.title}</h4>
        </div>
      </div>
    </div>
  )
}

export default class Budget extends Component {
  constructor(props) {
    super(props);
  }

  render(){
    return (
      <div>
      <main className="col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3">
        <h1>{metaProperties.pageTitle}</h1>
        {budgetWidgets.map((widget, i) => {
          return <BudgetWidgetCard title={widget} key={i} />
        })}
        <div className="card" style={{width: '65%', margin: 'auto', marginTop: '200px'}}>
          <h3 className="card-header">{metaProperties.budgetCardTitle}</h3>
          <div className="card-block" style={{padding: '20px'}}>
            <AddBudgetForm addBudget={this.props.addBudget} />
          </div>
        </div>
      </main>
      </div>
    )
  }
}
