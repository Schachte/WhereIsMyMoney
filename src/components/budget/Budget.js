import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import FullWidthCard from '../common/FullWidthCard';
import AddBudgetForm from './AddBudgetForm';
import { validateForm,  checkIfFormIsValid } from './validation/AddBudgetFormValidation';

class Budget extends Component {
  constructor(props) {
    super(props);
    this.state = {
      budget: Object.assign({}, props.budgetObject),
      errors:  {budgetCategory: '', budgetCost: '', budgetDate: ''}
    };
    this.updateBudgetFormAndAddressErrors = this.updateBudgetFormAndAddressErrors.bind(this);
    this.onBeforeSave = this.onBeforeSave.bind(this);
  }

  updateBudgetFormAndAddressErrors(event) {
    const userInput = event.target.value;
    const fieldName = event.target.name;
    let budget = this.state.budget;
    budget[fieldName] = userInput;
    this.setState({budget: budget});
    this.setState({errors: validateForm(fieldName, userInput, this.state)});
  }

  onBeforeSave(e, formData) {
    e.preventDefault();
    let formValidator = checkIfFormIsValid(formData, this.state);
    if (formValidator.valid == false) {
      this.setState({errors: formValidator.errors});
    }
  }

  render() {
    return (
      <div className="budget-form-component">
        <FullWidthCard
          name = "budget-form-card"
          title = " Add New Budget Category"
          iconName = "fa fa-credit-card"
          nestedComponent = {
            <AddBudgetForm
              budget={this.state.budget}
              onChange={this.updateBudgetFormAndAddressErrors}
              onSave={this.onBeforeSave}
              errors={this.state.errors}
            />
          }
        />
      </div>
    );
  }
}

Budget.propTypes = {
  actions: PropTypes.object,
  budgetObject: PropTypes.object,
  addBudget: PropTypes.func
};

export default Budget;
