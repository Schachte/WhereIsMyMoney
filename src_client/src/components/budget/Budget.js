import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import FullWidthCard from '../common/FullWidthCard';
import AddBudgetForm from './AddBudgetForm';
import RenderBudgetTable from './BudgetTable';

import { validateForm,  checkIfFormIsValid } from './validation/AddBudgetFormValidation';

class Budget extends Component {
  constructor(props) {
    super(props);
    this.state = {
      budget: Object.assign({}, props.budgetObject),
      errors: {budgetCategory: '', budgetCost: '', budgetDate: ''}
    };
    this.updateBudgetFormAndAddressErrors = this.updateBudgetFormAndAddressErrors.bind(this);
    this.onBeforeSave = this.onBeforeSave.bind(this);
  }

  updateBudgetFormAndAddressErrors(event) {
    const userInput = event.target.value;
    const fieldName = event.target.name;
    let budget = this.state.budget;
    budget[fieldName] = userInput;
    this.setState({budget: budget});
    this.setState({errors: validateForm(fieldName, userInput, this.state)});
  }

  onBeforeSave(e, formData) {
    e.preventDefault();
    let formValidator = checkIfFormIsValid(formData, this.state);
    if (formValidator.valid == false) {
      this.setState({errors: formValidator.errors});
    } else {
      this.props.actions.addBudget(formData);
    }
  }

  render() {
    return (
      <div className="budget-form-component">
        <FullWidthCard
          name = "budget-form-card"
          title = " Add New Budget Category"
          iconName = "fa fa-credit-card"
          nestedComponent = {
            <AddBudgetForm
              budget={this.state.budget}
              onChange={this.updateBudgetFormAndAddressErrors}
              onSave={this.onBeforeSave}
              errors={this.state.errors}
            />
          }
        />

        <FullWidthCard
          name = "budget-form-table"
          title = "View Currently Saved Budgets"
          iconName = "fa fa-credit-card"
          nestedComponent = {
            <RenderBudgetTable
              budgets={this.props.budgets}/>
          }
        />
      </div>
    );
  }
}

Budget.propTypes = {
  actions: PropTypes.object,
  budgetObject: PropTypes.object,
  addBudget: PropTypes.func,
  budgets: PropTypes.array
};

export default Budget;
