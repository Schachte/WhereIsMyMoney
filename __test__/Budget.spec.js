import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
// import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import {createStore} from 'redux'
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configureStore } from '../src/configuration/store';
configure({ adapter: new Adapter() });

/**
BUDGET COMPONENTS
**/
import Budget from '../src/Budget/components/Budget';
import { BudgetWidgetCard } from '../src/Budget/components/Budget';

let mainHeaderTitle = 'Review Current Budget';
const store = configureStore();
const wrapper = shallow(<Budget />);

// Snapshot for Budget React Component
describe('>>>B U D G E T --- Snapshot',()=>{
  it('+++capturing Snapshot of Budget', () => {
      const wrapper = shallow(<Budget />);
      expect(wrapper).toMatchSnapshot();
    })
});

describe('>>>B U D G E T --- Elements Getting Rendered Correctly',()=>{
  it('+++correct title is getting rendered as header', () => {
      expect(wrapper.find('h1[name="main-header-title"]').text()).toEqual(mainHeaderTitle);
    })

  it('+++renders the budget widget items correctly', () => {
      expect(wrapper.find(BudgetWidgetCard).length).toEqual(4);
    })
});
