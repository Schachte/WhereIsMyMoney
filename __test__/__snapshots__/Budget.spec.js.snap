// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>B U D G E T --- Snapshot +++capturing Snapshot of Budget 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Budget />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main
        className="col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3"
      >
        <h1
          name="main-header-title"
        >
          Review Current Budget
        </h1>
        <BudgetWidgetCard
          className="budget-widget"
          state={Object {}}
          title="Monthly Budget"
        />
        <BudgetWidgetCard
          className="budget-widget"
          state={Object {}}
          title="No. Budgets"
        />
        <BudgetWidgetCard
          className="budget-widget"
          state={Object {}}
          title="Amount Left"
        />
        <BudgetWidgetCard
          className="budget-widget"
          state={Object {}}
          title="Avg. Weekly Spent"
        />
        <div
          className="card"
          style={
            Object {
              "margin": "auto",
              "marginTop": "150px",
              "width": "65%",
            }
          }
        >
          <h3
            className="card-header"
          >
            Add A New Budget Item
          </h3>
          <div
            className="card-block"
            style={
              Object {
                "padding": "20px",
              }
            }
          >
            <ReduxForm
              addBudget={undefined}
            />
          </div>
        </div>
        <div>
          <BudgetTable
            addEditableFieldErrors={undefined}
            budgetFormEditable={undefined}
            budgetFormEditableErrors={undefined}
            reduxForm={undefined}
            updateBudgetEntry={undefined}
            updateFieldEnableRequest={undefined}
            userBudgetItems={undefined}
          />
        </div>
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            name="main-header-title"
          >
            Review Current Budget
          </h1>,
          Array [
            <BudgetWidgetCard
              className="budget-widget"
              state={Object {}}
              title="Monthly Budget"
            />,
            <BudgetWidgetCard
              className="budget-widget"
              state={Object {}}
              title="No. Budgets"
            />,
            <BudgetWidgetCard
              className="budget-widget"
              state={Object {}}
              title="Amount Left"
            />,
            <BudgetWidgetCard
              className="budget-widget"
              state={Object {}}
              title="Avg. Weekly Spent"
            />,
          ],
          <div
            className="card"
            style={
              Object {
                "margin": "auto",
                "marginTop": "150px",
                "width": "65%",
              }
            }
          >
            <h3
              className="card-header"
            >
              Add A New Budget Item
            </h3>
            <div
              className="card-block"
              style={
                Object {
                  "padding": "20px",
                }
              }
            >
              <ReduxForm
                addBudget={undefined}
              />
            </div>
          </div>,
          <div>
            <BudgetTable
              addEditableFieldErrors={undefined}
              budgetFormEditable={undefined}
              budgetFormEditableErrors={undefined}
              reduxForm={undefined}
              updateBudgetEntry={undefined}
              updateFieldEnableRequest={undefined}
              userBudgetItems={undefined}
            />
          </div>,
        ],
        "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Review Current Budget",
            "name": "main-header-title",
          },
          "ref": null,
          "rendered": "Review Current Budget",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "className": "budget-widget",
            "state": Object {},
            "title": "Monthly Budget",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "className": "budget-widget",
            "state": Object {},
            "title": "No. Budgets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "className": "budget-widget",
            "state": Object {},
            "title": "Amount Left",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "className": "budget-widget",
            "state": Object {},
            "title": "Avg. Weekly Spent",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="card-header"
              >
                Add A New Budget Item
              </h3>,
              <div
                className="card-block"
                style={
                  Object {
                    "padding": "20px",
                  }
                }
              >
                <ReduxForm
                  addBudget={undefined}
                />
              </div>,
            ],
            "className": "card",
            "style": Object {
              "margin": "auto",
              "marginTop": "150px",
              "width": "65%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add A New Budget Item",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Add A New Budget Item",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReduxForm
                  addBudget={undefined}
                />,
                "className": "card-block",
                "style": Object {
                  "padding": "20px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "addBudget": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BudgetTable
              addEditableFieldErrors={undefined}
              budgetFormEditable={undefined}
              budgetFormEditableErrors={undefined}
              reduxForm={undefined}
              updateBudgetEntry={undefined}
              updateFieldEnableRequest={undefined}
              userBudgetItems={undefined}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addEditableFieldErrors": undefined,
              "budgetFormEditable": undefined,
              "budgetFormEditableErrors": undefined,
              "reduxForm": undefined,
              "updateBudgetEntry": undefined,
              "updateFieldEnableRequest": undefined,
              "userBudgetItems": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main
          className="col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3"
        >
          <h1
            name="main-header-title"
          >
            Review Current Budget
          </h1>
          <BudgetWidgetCard
            className="budget-widget"
            state={Object {}}
            title="Monthly Budget"
          />
          <BudgetWidgetCard
            className="budget-widget"
            state={Object {}}
            title="No. Budgets"
          />
          <BudgetWidgetCard
            className="budget-widget"
            state={Object {}}
            title="Amount Left"
          />
          <BudgetWidgetCard
            className="budget-widget"
            state={Object {}}
            title="Avg. Weekly Spent"
          />
          <div
            className="card"
            style={
              Object {
                "margin": "auto",
                "marginTop": "150px",
                "width": "65%",
              }
            }
          >
            <h3
              className="card-header"
            >
              Add A New Budget Item
            </h3>
            <div
              className="card-block"
              style={
                Object {
                  "padding": "20px",
                }
              }
            >
              <ReduxForm
                addBudget={undefined}
              />
            </div>
          </div>
          <div>
            <BudgetTable
              addEditableFieldErrors={undefined}
              budgetFormEditable={undefined}
              budgetFormEditableErrors={undefined}
              reduxForm={undefined}
              updateBudgetEntry={undefined}
              updateFieldEnableRequest={undefined}
              userBudgetItems={undefined}
            />
          </div>
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              name="main-header-title"
            >
              Review Current Budget
            </h1>,
            Array [
              <BudgetWidgetCard
                className="budget-widget"
                state={Object {}}
                title="Monthly Budget"
              />,
              <BudgetWidgetCard
                className="budget-widget"
                state={Object {}}
                title="No. Budgets"
              />,
              <BudgetWidgetCard
                className="budget-widget"
                state={Object {}}
                title="Amount Left"
              />,
              <BudgetWidgetCard
                className="budget-widget"
                state={Object {}}
                title="Avg. Weekly Spent"
              />,
            ],
            <div
              className="card"
              style={
                Object {
                  "margin": "auto",
                  "marginTop": "150px",
                  "width": "65%",
                }
              }
            >
              <h3
                className="card-header"
              >
                Add A New Budget Item
              </h3>
              <div
                className="card-block"
                style={
                  Object {
                    "padding": "20px",
                  }
                }
              >
                <ReduxForm
                  addBudget={undefined}
                />
              </div>
            </div>,
            <div>
              <BudgetTable
                addEditableFieldErrors={undefined}
                budgetFormEditable={undefined}
                budgetFormEditableErrors={undefined}
                reduxForm={undefined}
                updateBudgetEntry={undefined}
                updateFieldEnableRequest={undefined}
                userBudgetItems={undefined}
              />
            </div>,
          ],
          "className": "col-sm-9 offset-sm-3 col-md-10 offset-md-2 pt-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Review Current Budget",
              "name": "main-header-title",
            },
            "ref": null,
            "rendered": "Review Current Budget",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "className": "budget-widget",
              "state": Object {},
              "title": "Monthly Budget",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "className": "budget-widget",
              "state": Object {},
              "title": "No. Budgets",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "className": "budget-widget",
              "state": Object {},
              "title": "Amount Left",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "className": "budget-widget",
              "state": Object {},
              "title": "Avg. Weekly Spent",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="card-header"
                >
                  Add A New Budget Item
                </h3>,
                <div
                  className="card-block"
                  style={
                    Object {
                      "padding": "20px",
                    }
                  }
                >
                  <ReduxForm
                    addBudget={undefined}
                  />
                </div>,
              ],
              "className": "card",
              "style": Object {
                "margin": "auto",
                "marginTop": "150px",
                "width": "65%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add A New Budget Item",
                  "className": "card-header",
                },
                "ref": null,
                "rendered": "Add A New Budget Item",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ReduxForm
                    addBudget={undefined}
                  />,
                  "className": "card-block",
                  "style": Object {
                    "padding": "20px",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "addBudget": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BudgetTable
                addEditableFieldErrors={undefined}
                budgetFormEditable={undefined}
                budgetFormEditableErrors={undefined}
                reduxForm={undefined}
                updateBudgetEntry={undefined}
                updateFieldEnableRequest={undefined}
                userBudgetItems={undefined}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addEditableFieldErrors": undefined,
                "budgetFormEditable": undefined,
                "budgetFormEditableErrors": undefined,
                "reduxForm": undefined,
                "updateBudgetEntry": undefined,
                "updateFieldEnableRequest": undefined,
                "userBudgetItems": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
